pages/api/load.ts

```diff
@@ -1,20 +1,81 @@
 import { NextApiRequest, NextApiResponse } from 'next';
+import { createAppExtension, getAppExtensions } from '@lib/appExtensions';
+import { getStoreToken } from '@lib/dbs/firebase';
 import { encodePayload, getBCVerify, setSession } from '../../lib/auth';
 
 const buildRedirectUrl = (url: string, encodedContext: string) => {
     const [path, query = ''] = url.split('?');
     const queryParams = new URLSearchParams(`context=${encodedContext}&${query}`);
 
     return `${path}?${queryParams}`;
-}
+};
 
 export default async function load(req: NextApiRequest, res: NextApiResponse) {
     try {
         // Verify when app loaded (launch)
         const session = await getBCVerify(req.query);
         const encodedContext = encodePayload(session); // Signed JWT to validate/ prevent tampering
 
+        const { sub } = session;
+
+        const storeHash = sub?.split('/')[1] || '';
+
+        // Get access token from DB
+        const accessToken = await getStoreToken(storeHash);
+
         await setSession(session);
+
+        /**
+         * For stores that do not have app extensions installed yet, create app extensions when app is
+         * loaded
+         */
+
+        const requestBody = await getAppExtensions();
+
+        const existingAppExtensions = await fetch(
+            `https://api.bigcommerce.com/stores/${storeHash}/graphql`,
+            {
+                method: 'POST',
+                headers: {
+                    accept: 'application/json',
+                    'content-type': 'application/json',
+                    'x-auth-token': accessToken
+                },
+                body: JSON.stringify(requestBody)
+            }
+        );
+
+        const existingAppExtensionsData = await existingAppExtensions.json();
+
+        // Check for any if app extensions exist, if not create them
+        if (!existingAppExtensionsData?.data?.store?.appExtensions?.edges?.length) {
+            const createAppExtensionRequest = await createAppExtension();
+
+            // Make fetch to create the app extensions
+            const createResponse = await fetch(
+                `https://api.bigcommerce.com/stores/${storeHash}/graphql`,
+                {
+                    method: 'POST',
+                    headers: {
+                        accept: 'application/json',
+                        'content-type': 'application/json',
+                        'x-auth-token': accessToken
+                    },
+                    body: JSON.stringify(createAppExtensionRequest)
+                }
+            );
+            const { data: createAppExtensionData, errors: createAppExtensionErrors } =
```

Comment:
qq: why are we renaming `data` and `errors` here? it doesn't seem like these two variables have been namespaced in this section already

Reply:
^ changing this I renamed this when I was making changes and had another error check with a second api call that I didn't need, this is unnecessary now as its just the one instance